### PR TITLE
NEXUS-4554, Stops walker from walking over Out of service repos

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,3 +32,4 @@ target/
 nexus/nexus-test-harness/nexus-test-harness-its/test-output
 nexus/nexus-clients/nexus-rest-client-java/test-output/
 
+nexus/nexus-core-plugins/nexus-migration-plugin/nexus-migration-plugin-test-harness/plexus.nexus-work_IS_UNDEFINED/logs/migration.log
```
Comment (peterlynch) [inline .gitignore]: wth is plexus.nexus-work_IS_UNDEFINED and can we make it defined?
I know this is not your problem - but gitignore will now be hiding the real problem and mvn clean won't clean it up on the grid

File: nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/walker/WalkerTest.java [hunk 2]
```diff
@@ -90,11 +97,45 @@ public class WalkerTest
...
+        Assert.assertEquals( 10, wp.collExits );
+        Assert.assertEquals( 10, wp.colls );
+        Assert.assertEquals( 4, wp.files );
+        Assert.assertEquals( 0, wp.links );
+    }
+
+    /**
+     * Tests walking an out of service repo.  The walker should NOT not fail, but also NOT find any items.
+     * @throws Exception
+     */
```
Comment (peterlynch) [inline nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/walker/WalkerTest.java:111]: might be helpful to add a ref to the related issue numbers here

File: nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/wastebasket/DefaultWastebasketTest.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        exRepoConf.setChecksumPolicy( ChecksumPolicy.STRICT_IF_EXISTS );
+
+        repo.configure( repoConf );
+
+        lookup( ApplicationConfiguration.class ).getConfigurationModel().addRepository( repoConf );
+
+        lookup( RepositoryRegistry.class ).addRepository( repo );
+    }
+
+    @Test
```
Comment (peterlynch) [inline nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/wastebasket/DefaultWastebasketTest.java:81]: an we add  the related issue #

File: nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/walker/DefaultWalker.java [hunk 4]
```diff
@@ -29,6 +29,7 @@ import org.sonatype.nexus.proxy.StorageException;
```
Comment (bdemers) [inline nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/walker/DefaultWalker.java:28]: Something is wrong here, there are extra diffs

File: nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/walker/WalkerTest.java (hunk 2 above)
Comment (bdemers) [inline nexus/nexus-proxy/src/test/java/org/sonatype/nexus/proxy/walker/WalkerTest.java:111]: Good idea

File: nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/walker/DefaultWalker.java [hunk 5]
```diff
@@ -56,74 +58,81 @@ public class DefaultWalker
             fromPath = RepositoryItemUid.PATH_ROOT;
         }
 
-        context.getContext().put( WALKER_WALKED_FROM_PATH, fromPath );
-
-        if ( getLogger().isDebugEnabled() )
+        // cannot walk out of service repos
+        if( LocalStatus.OUT_OF_SERVICE == context.getRepository().getLocalStatus() )
```
Comment (bdemers) [inline nexus/nexus-proxy/src/main/java/org/sonatype/nexus/proxy/walker/DefaultWalker.java:62]: The extra white space makes this diff almost impossible to understand.
Added block if( LocalStatus.OUT_OF_SERVICE == context.getRepository().getLocalStatus() )

**Comment by cstamas** [comment]
Accepted, same should be done with indexer.

**Pull request merged**